### PR TITLE
Implement part of intermine_boot

### PR DESCRIPTION
**Pull request by heralden**
~~A lot of work remains until this PR is ready. I created it early for tracking progress.~~

You can test this by following the README and invoking intermine_boot:
```
# Start (build) a biotestmine and stop it.
intermine_boot start local
intermine_boot stop local
# Build biotestmine using a custom build of InterMine. 
intermine_boot start local --build-im --im-repo https://github.com/danielabutano/intermine --im-branch bluegenes
# Add `--build-images` to build docker images locally instead of fetching from Docker Hub.
```
Not all of the CLI flags documented have effect yet.

### CODE REVIEW

File: intermine_boot/__init__.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import subprocess
+import sys
+import re
+import click
+from xdg import (XDG_DATA_HOME)
```
Comment (leoank) [inline intermine_boot/__init__.py:5]: Do we really need this dependency? Almost all unix systems have $HOME defined. We can just use `os.environ.get('HOME')` for home directory path.

File: intermine_boot/commands.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        else:
+            click.echo(out, err=True)
+            sys.exit(docker_info.returncode)
+
+
+def start(options, env):
+    assert_docker(options, env)
+    docker.up(options, env)
+    docker.monitor_builder(options, env)
+
```
Comment (leoank) [inline intermine_boot/commands.py:37]: A friendly message at the end that says something like "Build is finished, visit localhost:9999" will be very helpful. We can do it by adding a `click.echo()` statement here.

File: intermine_boot/docker.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
+from pathlib import Path
+import subprocess
+import shutil
+import os
+from git import Repo
+import yaml
+from intermine_boot import utils
+
+DOCKER_COMPOSE_REPO = 'https://github.com/uosl/docker-intermine-gradle'
```
Comment (leoank) [inline intermine_boot/docker.py]: Send a PR to main repo. Then change this.

**Review by leoank — COMMENTED** [review]
Looks good to me!

File: intermine_boot/__init__.py (hunk 1 above)
Comment (heralden) [reply intermine_boot/__init__.py:5]: Storing application files in dot files/dirs in $HOME is deprecated and replaced by the XDG spec (for Linux, XDG_DATA_HOME is `~/.local/share`). The dependency is pretty light, and I believe more cross-platform than reading the XDG env vars manually.
https://wiki.archlinux.org/index.php/XDG_Base_Directory

File: intermine_boot/commands.py (hunk 2 above)
Comment (heralden) [reply intermine_boot/commands.py:37]: Great idea! That will show the tomcat status page though, so I added /biotestmine with a note to change it once we support building other mines.

File: intermine_boot/docker.py (hunk 3 above)
Comment (heralden) [reply intermine_boot/docker.py]: Done!

File: intermine_boot/__init__.py (hunk 1 above)
Comment (leoank) [reply intermine_boot/__init__.py:5]: learned something new today! thanks

**Pull request merged**